requirements.txt

```diff
@@ -5,6 +5,8 @@ pytest==8.0.0
 pytest-xdist==3.5.0
 mypy==1.8.0
 types-requests==2.31.0.20240125
+huggingface-hub==0.22.2
+transformers==4.40.0
```

Comment:
I ran into this also.

Do we want these added at the top level here, or included optionally?
* huggingface-hub is used by [`sharktank/sharktank/utils/hf_datasets.py`](https://github.com/nod-ai/sharktank/blob/main/sharktank/sharktank/utils/hf_datasets.py)
* transformers is used by [`sharktank/sharktank/utils/tokenizer.py`](https://github.com/nod-ai/sharktank/blob/main/sharktank/sharktank/utils/tokenizer.py)